requires.py

```diff
@@ -97,6 +102,20 @@ def instance_id(self):
                 unitdata.kv().set(cache_key, self._instance_id)
         return self._instance_id
 
+    @property
+    def region(self):
+        if not hasattr(self, '_region'):
+            cache_key = self.expand_name('region')
+            cached = unitdata.kv().get(cache_key)
+            if cached:
+                self._region = cached
+            else:
+                with urlopen(self._az_url) as fd:
+                    az = fd.read(256).decode('utf8')
```

Comment:
why limit the read?

Reply:
Just to keep us from blindly reading a completely unlimited amount of data into memory.  *shrug*  I'd be ok bumping it up to 65k or something, but I'd rather it not just be completely open-ended, even though if that's an issue, we probably have bigger problems anyway.